flink-streaming-java/src/test/java/org/apache/flink/streaming/api/functions/sink/filesystem/TestUtils.java

```diff
@@ -414,4 +419,50 @@ public void clear() {
 			backingList.clear();
 		}
 	}
+
+	static class LocalRecoverableWriterForBucketStateMigrationTest extends LocalRecoverableWriter {
+
+		final String prefix = "src/test/resources/";
+
+		LocalRecoverableWriterForBucketStateMigrationTest() {
+			super(new LocalFileSystem());
+		}
+
+		public RecoverableFsDataOutputStream open(Path filePath) throws IOException {
```

Comment:
I think maybe we don't need this if we take the minor change from my branch that allows relative files: https://github.com/aljoscha/flink/commit/99810151e4dd37a5b61b89c8056a1ad1202d75a4